feat: add zkevm config

bilbeyt opened the pull request:
Closes: #1776

GabrielBuragev commented:
I think we could rename the files to `{chain-id}-polygon-zkevm.json` as `{chain-id}-zkevm.json` is a little bit too broad/general for a network name.

bilbeyt commented:
> I think we could rename the files to `{chain-id}-polygon-zkevm.json` as `{chain-id}-zkevm.json` is a little bit too broad/general for a network name.

I did the naming also for ape package to verify contracts. ape-zkevm which if i change here the name to polygon-zkevm my script cant verify the contract. I remember i tried to use this also like changing ape-zkevm to ape-polygon_zkevm or somethiing but it didnt work

istankovic commented:
> I think we could rename the files to `{chain-id}-polygon-zkevm.json` as `{chain-id}-zkevm.json` is a little bit too broad/general for a network name.

I think that's a good point by @GabrielBuragev, @bilbeyt is it too much work to change it now?

bilbeyt commented:
> > I think we could rename the files to `{chain-id}-polygon-zkevm.json` as `{chain-id}-zkevm.json` is a little bit too broad/general for a network name.
> 
> I think that's a good point by @GabrielBuragev, @bilbeyt is it too much work to change it now?

I changed the ape package now and changed config name. @istankovic @GabrielBuragev

istankovic commented on beamer/deploy/util.py:
```diff
@@ -85,22 +85,26 @@ def deploy_contract(web3: Web3, constructor_spec: Union[str, Sequence]) -> Deplo
+            args.append(request_manager.address)
+        else:
+            args.append(arg)
+    return tuple(args)
```
I think we should rather handle this in a generic manner at this point, something like

- instead of the `resolver` and the `request_manager` arguments, this function should take a dictionary `dict[str, Contract]`
- have code that does the `${...}` substitution by looking the key in the provided dict

istankovic reviewed (changes requested):
Left a comment on the deployment script part.
Please let someone else take a look at the config.

GabrielBuragev commented:
@istankovic I am currently doing a sanity check of the config file. Thanks for taking over the python part.

fredo commented on beamer/deploy/util.py:
```diff
@@ -111,6 +108,17 @@ def deploy_beamer(
+    args = _resolve_constructor_args(contract_args, chain.l2_messenger)
+    l2_messenger = deploy_contract(w3, args)
+
+    if "Polygon ZkEVM" in chain.name:
```
I would actually prefer to go with chain ids here as an identifier.

fredo commented:
Addressed just some nitpicks. Otherwise looks good.

GabrielBuragev reviewed (approved):
Config looks fine! Only one thing we need to clear up - finality_period

GabrielBuragev commented on deployments/config/mainnet/1101-polygon-zkevm.json:
```diff
@@ -0,0 +1,42 @@
+        "${request_manager}",
+        0
+    ],
+    "finality_period": 3600,
```
There is no precise `finality_period` for polygonZkevm?

fredo replied:
I bet there is, but I would just leave the value blank before we are sure what to choose, that was my rationale

bilbeyt commented on deployments/config/goerli/1442-polygon-zkevm.json:
```diff
@@ -0,0 +1,37 @@
+        "${request_manager}",
+        0
+    ],
+    "finality_period": 3600,
```
asked them in their discord also there is a tweet matching with the answer

bilbeyt replied:
https://twitter.com/0xPolygon/status/1638970957657174017?s=20

GabrielBuragev replied:
Great, thanks! Lets hope their plan is not to increase the finality periods :D
